From 9f9ec4224c96613e945fde977204f7f2368f5d89 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 30 Mar 2016 16:12:35 +0300
Subject: [PATCH 0134/1345] pci: msi: enable MSI interrupts on CP-110 pci

commit  39995ede93ac01ad17b82c8cec88ea125c7fbf98 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The change is made in 7040-A0 files since MSI interrupts
are not supported in AP806-Z1

Change-Id: I13f9c8af0d1a4b8e9d4ca071be3d2eb3356a067e
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28670
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi       |   12 ++++++++++++
 .../boot/dts/marvell/armada-cp110-master.dtsi      |    3 ---
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 78d995d..40e644f 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -52,4 +52,16 @@
 	model = "Marvell Armada 7040";
 	compatible = "marvell,armada7040", "marvell,armada-ap806-quad",
 		     "marvell,armada-ap806";
+
+	cp110-master {
+		pcie@0x600000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x620000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x640000 {
+			msi-parent = <&gic_v2m0>;
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
index 76d6242..c572f96 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
@@ -335,7 +335,6 @@
 			#size-cells = <2>;
 			#interrupt-cells = <1>;
 			device_type = "pci";
-			/* msi-parent = <&gic_v2m1>; */
 			dma-coherent;
 
 			bus-range = <0 0xff>;
@@ -357,7 +356,6 @@
 			#size-cells = <2>;
 			#interrupt-cells = <1>;
 			device_type = "pci";
-			/* msi-parent = <&gic_v2m1>; */
 			dma-coherent;
 
 			bus-range = <0 0xff>;
@@ -380,7 +378,6 @@
 			#size-cells = <2>;
 			#interrupt-cells = <1>;
 			device_type = "pci";
-			/* msi-parent = <&gic_v2m2>; */
 			dma-coherent;
 
 			bus-range = <0 0xff>;
-- 
1.7.9.5

